clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/controllers/AmazonNamedImageLookupController.groovy

```diff
@@ -95,11 +94,65 @@ class AmazonNamedImageLookupController {
 
     Collection<CacheData> matchesByImageId = cacheView.getAll(IMAGES.ns, imageIdentifiers)
 
-    render(matchesByName, matchesByImageId, null, lookupOptions.q, lookupOptions.region)
+    return filter(
+      render(matchesByName, matchesByImageId, lookupOptions.q, lookupOptions.region),
+      request
+    )
+  }
+
+  /**
+   * Apply tag-based filtering to the list of named images.
+   *
+   * ie. /aws/images/find?q=PackageName&tag:engine=spinnaker&tag:stage=released
+   */
+  private List<NamedImage> filter(List<NamedImage> namedImages, HttpServletRequest httpServletRequest) {
```

Comment:
I'd probably pull the request parameter parsing out of here and pass in the tag filters as an arg to this method